pkg/param/binary_params.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2023 PingCAP, Inc.
```

Comment:
Most of this file is copied from `/pkg/internal/parse`. Because it's not expected to be "internal" inside the `server`.